dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1437,6 +1442,20 @@ public List<Map<String, Object>> getContentletReferences(final Contentlet conten
         return results;
     }
 
+    @CloseDBIfOpened
+    @Override
+    public Optional<Integer> getContentletReferenceCount(final String contentletId) {
+        try {
+            return Optional.of(this.multiTreeAPI.get().getContentletReferenceCount(contentletId));
+        } catch (final DotDataException e) {
+            final String errorMsg = String.format("An error occurred when getting the page references for Contentlet " +
+                                                          "ID '%s': %s", contentletId, e.getMessage());
+            Logger.warn(this, errorMsg);
+            Logger.debug(this, errorMsg, e);
```

Comment:
We should throw exception in case of errors... instead of returning Optional.empty